src/legacy/core_plugins/dashboard_embeddable_container/public/np_ready/public/tests/dashboard_container.test.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { findTestSubject } from '@elastic/eui/lib/test';
+import React from 'react';
+import { mount } from 'enzyme';
+import { nextTick } from 'test_utils/enzyme_helpers';
+import { I18nProvider } from '@kbn/i18n/react';
+import { ViewMode, CONTEXT_MENU_TRIGGER, EmbeddablePanel } from '../lib/embeddable_api';
+import { DashboardContainer } from '../lib/embeddable/dashboard_container';
+import { getSampleDashboardInput } from '../lib/test_helpers';
+import {
+  CONTACT_CARD_EMBEDDABLE,
+  ContactCardEmbeddableFactory,
+} from '../../../../../embeddable_api/public/np_ready/public/lib/test_samples/embeddables/contact_card/contact_card_embeddable_factory';
+import {
+  ContactCardEmbeddableInput,
+  ContactCardEmbeddable,
+  ContactCardEmbeddableOutput,
+} from '../../../../../embeddable_api/public/np_ready/public/lib/test_samples/embeddables/contact_card/contact_card_embeddable';
+import { embeddablePluginMock } from '../../../../../embeddable_api/public/np_ready/public/mocks';
+import { EditModeAction } from '../../../../../embeddable_api/public/np_ready/public/lib/test_samples/actions/edit_mode_action';
+
+test('DashboardContainer in edit mode shows edit mode actions', async () => {
+  const { setup, doStart } = embeddablePluginMock.createInstance();
+
+  const editModeAction = new EditModeAction();
+  setup.registerAction(editModeAction);
+  setup.attachAction(CONTEXT_MENU_TRIGGER, editModeAction.id);
+  setup.registerEmbeddableFactory(
+    CONTACT_CARD_EMBEDDABLE,
+    new ContactCardEmbeddableFactory({} as any, (() => null) as any, {} as any)
+  );
+
+  const start = doStart();
+
+  const container = new DashboardContainer(getSampleDashboardInput({ viewMode: ViewMode.VIEW }), {
+    getEmbeddableFactory: start.getEmbeddableFactory,
+  } as any);
+
+  const embeddable = await container.addNewEmbeddable<
+    ContactCardEmbeddableInput,
+    ContactCardEmbeddableOutput,
+    ContactCardEmbeddable
+  >(CONTACT_CARD_EMBEDDABLE, {
+    firstName: 'Bob',
+  });
+
+  const component = mount(
+    <I18nProvider>
+      <EmbeddablePanel
+        embeddable={embeddable}
```

Comment:
this makes EmbeddablePanel way harder to use. Shouldn't we inject things into this component and export one without all this dependencies as part of our start contract ?

Reply:
---

Agreed to implement this in one of the following Embeddable PRs.